build.gradle.kts

```diff
@@ -79,6 +76,21 @@ tasks {
         kotlinOptions {
             allWarningsAsErrors = true
             jvmTarget = targetJavaVersion.toString()
+            languageVersion = targetKotlinVersion
+            apiVersion = targetKotlinVersion
+            if (JavaVersion.current().isJava9Compatible) {
+                freeCompilerArgs += "-Xjdk-release=1.8"
+            }
+        }
+    }
+
+    withType<JavaCompile> {
+        if (JavaVersion.current().isJava9Compatible) {
+            doFirst {
+                options.compilerArgs = listOf("--release", "8")
```

Comment:
Compiles with the release flag for any Java code to ensure no APIs release in Java 9+ are used.